CMakeLists.txt

```diff
@@ -253,31 +253,150 @@ endif()
 
 include_directories(${NANOGUI_EIGEN_INCLUDE_DIR} ext/glfw/include ext/nanovg/src include ${CMAKE_CURRENT_BINARY_DIR})
 
+# Parent projects set this to include additional *icon* fonts.  The expectation
+# is that this list is populated with the TTF font, and the associated header
+# file defining the available constants (and python bindings if applicable) are
+# in the *SAME* directory.  The naming must be exact:
+#
+#     /some/path/iconfont.ttf           <- list(APPEND NANOGUI_EXTRA_ICON_RESOURCES /some/path/iconfont.ttf)
+#     /some/path/iconfont.h             <- defines all icon codes (needed for C++ side)
+#     /some/path/constants_iconfont.cpp <- defines the python bindings (if NANOGUI_BUILD_PYTHON is ON)
+#
+# The iconfont.ttf will be embedded like all other fonts, the iconfont.h will
+# *automatically* be included (and installed, if applicable) with the main
+# NanoGUI code-base.  With /some/path/iconfont.ttf, the suffix is removed and
+# the file iconfont.h is searched for in the same directory.  If building Python,
+# the file constants_iconfont.cpp will be searched for, and parsed to make sure
+# that it has a signature exactly matching: void register_iconfont(py::module &m).
+# If all checks are successful, then the icon font will be included with
+# NANOGUI_EXTRA_RESOURCES, the header file will be copied to
+# ${CMAKE_CURRENT_BINARY_DIR}/nanogui (which is included with NANOGUI_EXTRA_INCS),
+# and if python is being built, the configurations for python/main.cpp.in will
+# be coordinated to bind the font for python.
+if (NANOGUI_EXTRA_ICON_RESOURCES)
+  foreach(icon_font ${NANOGUI_EXTRA_ICON_RESOURCES})
+    # Make sure the font is a true type font
+    get_filename_component(icon_font_ext "${icon_font}" EXT)
+    if (NOT "${icon_font_ext}" STREQUAL ".ttf")
```

Comment:
what about OTF?

Reply:
Good point!

Reply:
@wjakob how much do you care about `.otf`?  [NanoVG does not support them](https://github.com/memononen/nanovg/issues/37).  The issue describes an alternative, but I think the changes required to support it would be much further reaching than just using [fontstash](https://github.com/memononen/fontstash) to load fonts.  I think we'd have to update **all** code drawing fonts to use that over NanoVG...